### PR TITLE
feat(cdp): Add mailgun destination

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We were discussing if we could already do some limited email sending stuff, we just needed to support a few things...

Well I felt like adding the whole thing so here we are 😅 

## Changes

- [x] Adds generic `EmailTemplater` inputs for generating a template nicely 
- [x] Adds mailgun as the first supported integration using it
- [x] Email values are templated out in plugin-server as well
- [x] Fix the bytecode generation for the html (css {} characters get in the way) 

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: frontend/src/scenes/pipeline/hogfunctions/email-templater/emailTemplaterLogic.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        },
+    })),
+])
+
+function escapeHTMLStringCurlies(htmlString: string): string {
+    const parser = new DOMParser()
+    const doc = parser.parseFromString(htmlString, 'text/html')
+
+    function escapeCurlyBraces(text: string): string {
+        return text.replace(/{/g, '\\{')
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/scenes/pipeline/hogfunctions/email-templater/emailTemplaterLogic.tsx:98]: This seems like a false positive. We're not trying to escape backslashes 🤷‍♂️. We're also doing all that the "show more details" text says to do for global matching, so dunno. Dismiss?

File: posthog/cdp/templates/mailgun/template_mailgun.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    name="Send an email via Mailgun",
+    description="Send emails using the Mailgun HTTP API",
+    icon_url="/static/services/mailgun.png",
+    hog="""
+if (empty(inputs.template.to)) {
+    return false
+}
+
+fn multiPartFormEncode(data) {
+    let boundary := f'---011000010111000001101001'
```
Comment (mariusandra) [inline posthog/cdp/templates/mailgun/template_mailgun.py:19]: might be worth generating a random uuid here?

**Review by mariusandra — APPROVED** [review]
Looks good (...enough to demo at least)! I didn't try the actual API call though.

File: posthog/cdp/templates/mailgun/template_mailgun.py (hunk 2 above)
Comment (benjackwhite) [reply posthog/cdp/templates/mailgun/template_mailgun.py:19]: yeah i originally did but it didn't work for some reason

Comment (benjackwhite) [reply posthog/cdp/templates/mailgun/template_mailgun.py:19]: The spec says it has to be a specific length i think

**Pull request merged**